### PR TITLE
Allow disabled_features alongside classic or unset solution on Kibana spaces

### LINKED ISSUE #3201: [Bug] elasticstack_kibana_space: Invalid Attribute Combination after plan when disabled_features and solution are specified
**Describe the bug**
`elasticstack_kibana_space` produces `Error: Invalid Attribute Combination` on plan when resource contains keys `disabled_features` and `solution` at the same time.

In Elasticsearch it is allowed to use both of these mappings. I would add a warning however, with Kibana API it is possible to use any of the available values of `solution` and `disabled_features`. But in reality, if the value of `solution` is anything else than `classic`, the API allows to update the space, but the values of `disabled_features` are override with the default list silently. Therefore I suppose that this error is a bug when the .tf has `solution = "classic"`. Note that in the GUI, we can only configure the `disabled_features` when `solution` is set to `classic` (because solution is mandatory to set).

**To Reproduce**
1. TF configuration used

```hcl
resource "elasticstack_kibana_space" "test" {
  color       = "#FFFFFF"
  description = "test space"
  disabled_features = [
    "ml"
  ]
  name     = "test"
  space_id = "test"
  solution = "classic"
}
```

2. Run `terraform plan`
3. See the error in the output

```
╷
│ Error: Invalid Attribute Combination
│ 
│   with elasticstack_kibana_space.test,
│   on test.tf line 9, in resource "elasticstack_kibana_space" "test":
│    9:   solution = "classic"
│ 
│ Attribute "disabled_features" cannot be specified when "solution" is specified
╵
```
 
**Versions (please complete the following information):**
 - OS: Linux
 - Terraform OpenTofu v1.11.6
 - Provider version 0.15.2
 - Elasticsearch Version 9.1.2

### PR DESCRIPTION
**Pull request by tobio**
## Changelog
Customer impact: fix
Summary: Allow `elasticstack_kibana_space.disabled_features` be set when `solution` is `classic`, unset, or unknown.

## Detailed changes

Implements the OpenSpec change `allowed-if-dependent-null-support`.

Fixes https://github.com/elastic/terraform-provider-elasticstack/issues/3201

### Validator changes (`internal/utils/validators/conditional.go`)
- Add a required `AllowedIfOptions` argument to the `AllowedIfDependent*` constructors so each call site explicitly opts in (or not) to allowing a null/unset dependent path via `AllowNullDependent`.
- Treat an unknown dependent value as satisfying the dependency condition for `AllowedIf...` validators. `RequiredIf...` and `ForbiddenIf...` semantics are unchanged.
- Render null dependents as `<unset>` instead of `""` in `AllowedIf` validation error messages.
- Add `ValidateSet` to `Condition` so the new validator can be attached to `disabled_features` (a `SetAttribute`).
- Cover the new behavior with unit tests for the static-path and path-expression variants (matching / null-allowed / null-not-allowed / unknown / non-matching) plus a friendlier-error-message test.

### Kibana space resource (`internal/kibana/spaces/`)
- Replace the reciprocal `ConflictsWith` validators between `disabled_features` and `solution` with a single `AllowedIfDependentPathEquals(solution, "classic", AllowedIfOptions{AllowNullDependent: true})` on `disabled_features`.
- Accept `disabled_features` when `solution` is `classic`, when `solution` is omitted, or when `solution` is unknown during plan-time validation; reject it only when `solution` has a known non-`classic` value.
- Add `TestAccResourceSpace_disabledFeaturesSolutionValidation` with `PlanOnly` steps for the three required scenarios (omitted, classic, security rejected with `ExpectError`).
- Update the stale `ModifyPlan` comment that previously referenced the removed `ConflictsWith` validator.

### Other call sites
Updated to pass `AllowedIfOptions{}` (`AllowNullDependent: false`), preserving prior null behavior:
- `internal/fleet/output/schema.go`
- `internal/kibana/dashboard/schema.go`, `internal/kibana/dashboard/panelkit/panel_config_block.go`
- `internal/elasticsearch/ml/datafeed/schema.go`

Note: the broader "unknown dependent is now allowed" behavior applies to every `AllowedIf...` call site (not just Kibana spaces). This is intentional per the change design — it aligns plan-time validation with Terraform's computed-attribute semantics.

### Tests

- `go test ./internal/utils/validators/...` — passes.
- `TF_ACC=1 go test -v -run 'TestAccResourceSpace|TestAccResourceSpace_disabledFeaturesSolutionValidation' ./internal/kibana/spaces/...` — passes against a local 8.x stack.
- `make lint` / `make build` — clean.
- `openspec validate allowed-if-dependent-null-support --strict` — passes.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: internal/utils/validators/conditional.go [hunk 1]
```diff
@@ -169,20 +197,40 @@ func (v Condition) checkStaticPath(ctx context.Context, config tfsdk.Config) (bo
 //
```
[the remark this answers is not in the document]
Comment (tobio) [reply internal/utils/validators/conditional.go:197]: Addressed in fe3abc2e: updated dependentFieldHasAllowedValue doc comment to describe dependentEvaluation return type and null/unknown AllowedIf behavior.
